src/components/thread/ThreadSidebar.tsx

```diff
@@ -25,7 +26,16 @@ export interface ThreadSidebarProps {
 
 const ThreadSidebar: React.FC<ThreadSidebarProps> = (props) => {
   const { forceHideIdentity } = useForceHideIdentity();
-  const { threadRoot, categories, contentNotices } = useThreadContext();
+  const { threadRoot, categories, contentNotices, parentBoardSlug, threadId } = useThreadContext();
+  const { getLinkToThread } = useCachedLinks();
+
+    const linkToThread = getLinkToThread({
+// @ts-ignore
```

Comment:
How do I deal with this crime?

Reply:
I think the easiest way is to get the slug and thread id from `usePageDetails` instead, like this:

```ts
  const {threadId, slug} = usePageDetails<ThreadPageDetails>();
```

This will fetch them from the URL instead of the thread context, which is guaranteeed not to be null. We probably want to change `useThreadContext` so that at least threadId is never null, since I don't see how that could happen, but I'm not 100% sure why it is like that in the first place, and I suspect it is some tech debt I've been carrying around forever and that i'll hopefully work on removing soon.